beacon-chain/blockchain/fork_choice.go

```diff
@@ -363,53 +369,63 @@ func VoteCount(block *pb.BeaconBlock, state *pb.BeaconState, targets map[uint64]
 //        return None
 //    else:
 //        return get_ancestor(store, store.get_parent(block), slot)
-func BlockAncestor(block *pb.BeaconBlock, slot uint64, beaconDB *db.BeaconDB) (*pb.BeaconBlock, error) {
-	if block.Slot == slot {
-		return block, nil
+func BlockAncestor(targetBlock *pb.AttestationTarget, slot uint64, beaconDB *db.BeaconDB) ([]byte, error) {
+	if targetBlock.Slot == slot {
+		return targetBlock.BlockRoot[:], nil
 	}
-	if block.Slot < slot {
+	if targetBlock.Slot < slot {
 		return nil, nil
 	}
-	parentHash := bytesutil.ToBytes32(block.ParentRootHash32)
-	parent, err := beaconDB.Block(parentHash)
+	parentRoot := bytesutil.ToBytes32(targetBlock.ParentRoot)
+	parent, err := beaconDB.Block(parentRoot)
```

Comment:
can we get parent from map instead of reading from db?

Reply:
Ah unfortunately not in this function, as the ancestor lookup can be different than just fetching the parent